src/cli/command-processor.ts

```diff
@@ -0,0 +1,242 @@
+ */
+export class ExecuteMessage implements IntentMessage {
+  intentType = 'EXECUTE';
+  requestId: string;
+  localDeviceId: string;
+  executeCommand: string;
+  params: Record<string, unknown>;
+  customData: Record<string, unknown>;
+
+  /**
+   * @param requestId  The request id for a triggered Execute request.
+   * @param localDeviceId  The localDeviceId for a triggered Execute request.
+   * @param executeCommand  The single command string to set in the triggered Execute request.
+   * @param params  The params array for the single Execute command.
+   * @param customData  The customData array for the single Execute command.
+   * @returns  A new ExecuteCommand instance.
+   */
+  constructor(
+    requestId: string,
+    localDeviceId: string,
+    executeCommand: string,
+    params: Record<string, unknown>,
+    customData: Record<string, unknown>
+  ) {
+    this.requestId = requestId;
+    this.localDeviceId = localDeviceId;
+    this.executeCommand = executeCommand;
+    this.params = params;
+    this.customData = customData;
+  }
+}
+
+export class CommandProcessor {
+  private worker: Worker;
+  constructor(worker: Worker) {
+    this.worker = worker;
+  }
+
+  // Wrapper for argument parsing with yargs.
+  private async parseIntent(userCommand: string): Promise<IntentMessage> {
+    const argv = await yargs()
+      .option('intent_type', {
+        describe: 'The Intent type',
+        type: 'string',
+        demandOption: true,
+      })
+      .option('request_id', {
+        describe: 'The request Id',
+        type: 'string',
+        demandOption: true,
+      })
+      .option('discovery_buffer', {
+        describe: 'The IDENTIFY dicovery buffer represented as a string',
+        type: 'string',
+        demandOption: false,
+      })
+      .option('device_id', {
+        describe: 'The device Id',
+        type: 'string',
+        demandOption: false,
+      })
+      .option('local_device_id', {
+        describe: 'The local device Id',
+        type: 'string',
+        demandOption: false,
+      })
+      .option('command', {
+        describe: 'The execute command to send to device_id',
+        type: 'string',
+        demandOption: false,
+      })
+      .option('params', {
+        describe: 'The params argument for the execute command, in JSON format',
+        type: 'string',
+        demandOption: false,
+      })
+      .option('custom_data', {
+        describe:
+          'The customData argument for the execute command, in JSON format',
+        type: 'string',
+        demandOption: false,
+      })
+      .parse(userCommand, {}, (error, argv) => {
+        if (error !== null) {
+          throw error;
+        }
+        return argv;
+      });
+
+    /**
+     * Validate arguments and return an IntentMessage
+     */
+    switch (argv.intent_type) {
+      case 'IDENTIFY':
+        if (argv.discovery_buffer === undefined) {
```

Comment:
Is it possible to make `identify` and `execute` separate commands, so that we can require options for each subcommand?

ie. `local-cli identify --discovery_buffer=...`

Reply:
I agree this would clean the parsing up.  Will do.

Reply:
Done.